zio-cache/shared/src/main/scala/zio/cache/ManagedCache.scala

```diff
@@ -0,0 +1,388 @@
+package zio.cache
+
+import zio.ZManaged.Finalizer
+import zio._
+import zio.internal.MutableConcurrentQueue
+
+import java.time.{Clock, Duration, Instant}
+import java.util
+import java.util.concurrent.atomic.{AtomicBoolean, AtomicInteger, LongAdder}
+import scala.jdk.CollectionConverters._
+
+sealed abstract class ManagedCache[-Key, +Error, +Value] {
+
+  /**
+   * Returns statistics for this cache.
+   * @return
+   */
+  def cacheStats: UIO[CacheStats]
+
+  /**
+   * Return whether a Managed associated with the specified key exists in the cache.
```

Comment:
It isn't whether a `Managed` exists but whether a value exists, right?

Reply:
Right but that is the same as the normal `Cache` and inherent to it being an asynchronous cache.